#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Gen Add (NAME, Datatype, Bitmask LEVEL, Description, Default Value, Min Value, Max Value)
gen.add("distance_threshold", double_t, 0, "Distance Threshold Param", 0.1, 0.01, 10)
gen.add("axis", str_t, 0, "Perpendicular Planar Axis", "Z")
gen.add("epsilon_angle", double_t, 0, "Planar Segmentation Angle Uncertainty", 0.0, 0.0, 90.0)
gen.add("distance_above_plane", double_t, 0, "Crop Distance Above Plane", 0.005, 0.001, 1.0)
gen.add("normal_distance_weight", double_t, 0, "Normal Distance Weight", 0.1, 0.01, 1.0)
gen.add("max_iterations", double_t, 0, "Max Iterations", 100, 50, 20000)
gen.add("radius_limit", double_t, 0, "Cylinder Radius Limit", 0.1, 0.1, 2.0)
gen.add("cluster_tolerance", double_t, 0, "Euclidean Clustering Tolerance", 0.06, 0.001, 1.0)
gen.add("min_cluster_size", double_t, 0, "Min Cluster Size", 200, 10, 1000)
gen.add("max_cluster_size", double_t, 0, "Max Cluster Size", 10000, 1000, 30000)

exit(gen.generate(PACKAGE, "perception", "Segmentation"))